<launch>
    <arg name="namespace_prefix" default="demo_handeyecalibration" />
    <arg name="start_simulator" default="true" />

    <!-- tracking simulation frequency and noise parameters -->
    <param name="frequency" value="10" />
    <param name="translation_noise_stdev" value="0.001" />
    <param name="rotation_noise_stdev" value="0.0001" />

    <arg name="eye_on_hand" default="false" />

    <!-- transformations for the eye-on-base case -->
    <arg unless="$(arg eye_on_hand)" name="calibration_transformation" default="1 0 0.5 0 0 0 1" />
    <arg unless="$(arg eye_on_hand)" name="arbitrary_transformation" default="0.12 0.21 0.137 0 0 0 1" />
    <!-- transformations for the eye-on-hand case -->
    <arg if="$(arg eye_on_hand)" name="calibration_transformation" default="0.12 0.21 0.137 0 0 0 1" />
    <arg if="$(arg eye_on_hand)" name="arbitrary_transformation" default="1 0 0.5 0 0 0 1" />

    <arg name="robot_base_frame" value="iiwa_link_0" />
    <arg name="robot_effector_frame" value="iiwa_link_ee" />
    <arg name="tracking_base_frame" value="tracking_origin" />
    <arg name="tracking_marker_frame" value="tracking_marker" />

    <!-- start the robot -->
    <include file="$(find iiwa_moveit)/launch/demo.launch">
        <arg name="rviz" value="false" />
    </include>


    <!-- start easy_handeye -->
    <include ns="/iiwa" file="$(find easy_handeye)/launch/calibrate.launch" >
        <arg name="start_rviz" value="false" />

        <arg name="namespace_prefix" value="$(arg namespace_prefix)" />
        <arg name="eye_on_hand" value="false" />
        <arg name="publish_dummy" value="false" /> <!-- in our case it is done by the tracking simulator -->

        <arg name="tracking_base_frame" value="$(arg tracking_base_frame)" />
        <arg name="tracking_marker_frame" value="$(arg tracking_marker_frame)" />
        <arg name="robot_base_frame" value="$(arg robot_base_frame)" />
        <arg name="robot_effector_frame" value="$(arg robot_effector_frame)" />

        <arg name="freehand_robot_movement" value="false" />
        <arg name="robot_velocity_scaling" value="0.5" />
        <arg name="robot_acceleration_scaling" value="0.2" />
    </include>

    <!-- start rviz with custom configuration -->
    <node ns="/iiwa" name="$(anon rviz)" pkg="rviz" type="rviz" respawn="true"
          args="-d $(find easy_handeye_demo)/launch/iiwa_stack_config.rviz" output="screen" />

    <!-- start the simulated tracking -->
    <node name="tracking_simulator" pkg="easy_handeye_demo" type="tracking_simulator.py" output="screen">
        <param name="start_simulator" value="$(arg start_simulator)" />

        <param name="eye_on_hand" value="$(arg eye_on_hand)" />

        <param name="calibration_transformation" value="$(arg calibration_transformation)" />
        <param name="arbitrary_transformation" value="$(arg arbitrary_transformation)" />

        <param name="tracking_base_frame" value="$(arg tracking_base_frame)" />
        <param name="tracking_marker_frame" value="$(arg tracking_marker_frame)" />
        <param name="robot_base_frame" value="$(arg robot_base_frame)" />
        <param name="robot_effector_frame" value="$(arg robot_effector_frame)" />

        <param name="frequency" value="10" />
        <param name="translation_noise_stdev" value="0.0001" />
        <param name="rotation_noise_stdev" value="0.00001" />
    </node>
</launch>